MetaMask/metamask-extension
# Add setApprovalForAll confirmation view

Fixes: #15070
I haven't had time or any design feedback to create a wholly separate flow specific to `setApproveForAll`, rather I've further parameterized the `confirm-approve` component to provide specific verbiage for this method (both granting and revoking approval). This is not the way I would do it with more time, but the timeline has been compressed and [there is some urgency to get something out there](https://consensys.slack.com/archives/GTQAGKY5V/p1656003466814319?thread_ts=1655303420.735759&cid=GTQAGKY5V) since this method is so commonly used.

cc @rachelcope and/or @SayaGT we may want to tweak the verbiage here.

To Do after this PR:

`setApproveForAll` (when granting permission)
**Before:**
<img width="419" alt="Screen Shot 2022-06-28 at 4 35 39 PM" src="https://user-images.githubusercontent.com/34557516/176302848-f476bebb-4464-4563-98c7-1c0e5e73e2fe.png">

**After:**
<img width="545" alt="Screen Shot 2022-06-28 at 4 05 34 PM" src="https://user-images.githubusercontent.com/34557516/176295985-be7915aa-c914-473d-a458-8d6869849aec.png">
<img width="509" alt="Screen Shot 2022-06-28 at 4 04 57 PM" src="https://user-images.githubusercontent.com/34557516/176296162-d230b27c-70e5-4a01-9dcb-3f9580960bd8.png">

`setApproveForAll` (when revoking permission):

**Before:**
<img width="331" alt="Screen Shot 2022-06-28 at 4 36 05 PM" src="https://user-images.githubusercontent.com/34557516/176302871-a48b3aa7-a854-40ce-8902-ccb4b4ac2b3a.png">

**After:**
<img width="560" alt="Screen Shot 2022-06-28 at 4 25 50 PM" src="https://user-images.githubusercontent.com/34557516/176300684-6ec21643-ff8d-445b-8fee-c772eb673c01.png">
<img width="531" alt="Screen Shot 2022-06-28 at 4 26 24 PM" src="https://user-images.githubusercontent.com/34557516/176300731-cc186846-8830-488a-9831-ba7694106736.png">


**How to test:**
1. Go to etherscan and navigate to any ERC721 or ERC1155 contract
2. Navigate to the `contract` tab and then `write contract`:
<img width="751" alt="Screen Shot 2022-06-28 at 4 37 02 PM" src="https://user-images.githubusercontent.com/34557516/176303112-4e606385-8bd6-4a25-92d9-e23022beb3a6.png">
 3. Connect your wallet by clicking `Connect to Web3`
<img width="202" alt="Screen Shot 2022-06-28 at 4 38 08 PM" src="https://user-images.githubusercontent.com/34557516/176303480-cd4ac719-116c-428f-97a6-177f9f343785.png">
4. Scroll down to `setApprovalForAll` and click it so it drops down inputs
5. Enter args: For `operator` enter another address (any will do) and for `approved` enter true to see the grant permission flow. (then repeat and enter false to see the revoke flow)
